plugins/scanners/fossid/src/main/kotlin/events/CloneRepositoryHandler.kt

```diff
@@ -56,6 +56,26 @@ import org.semver4j.Semver
  * An event handler when FossID clones a repository itself.
  */
 class CloneRepositoryHandler(val config: FossIdConfig, val service: FossIdServiceWithVersion) : EventHandler {
+
+    companion object {
+        // Deduplicates and normalizes ignore rules by removing duplicate directory and file rules,
+        // and by stripping the trailing '/*' from directory rule values.
+        fun deduplicateAndNormalizeIgnoreRules(allRules: List<IgnoreRule>): List<IgnoreRule> {
```

Comment:
This should probably rather be a top-level `internal` function to not add it to the public API of this module.